src/compass/utils/geogrid.py

```diff
@@ -0,0 +1,361 @@
+    geo_grid: isce3.product.bbox_to_geogrid
+        ISCE3 object definining the geogrid
+    x_snap: float
+        Snap value along X-direction
+    y_snap: float
+        Snap value along Y-direction
+    x_end: float
+        Bottom-right X coordinate
+    y_end: float
+        Bottom-right Y coordinate
+
+    Returns
+    -------
+    geo_grid: isce3.product.bbox_to_geogrid
+        ISCE3 object containing the snapped geogrid
+    '''
+    if x_end is None: x_end = geo_grid.end_x
+    if y_end is None: y_end = geo_grid.end_y
+
+    if x_snap is not None or y_snap is not None:
+        snap_coord = lambda val, snap, round_func: round_func(
+            float(val) / snap) * snap
+        geo_grid.start_x = snap_coord(geo_grid.start_x, x_snap, np.floor)
+        geo_grid.start_y = snap_coord(geo_grid.start_y, y_snap, np.ceil)
+        end_x = snap_coord(x_end, x_snap, np.ceil)
+        end_y = snap_coord(y_end, y_snap, np.floor)
+        geo_grid.length = _grid_size(end_y, geo_grid.start_y,
+                                     geo_grid.spacing_y)
+        geo_grid.width = _grid_size(end_x, geo_grid.start_x, geo_grid.spacing_x)
+    return geo_grid
+
+
+def get_point_epsg(lat, lon):
+    '''
+    Get EPSG code based on latitude and longitude
+    coordinates of a point
+
+    Parameters
+    ----------
+    lat: float
+        Latitude coordinate of the point
+    lon: float
+        Longitude coordinate of the point
+
+    Returns
+    -------
+    epsg: int
+        UTM zone
+    '''
+    error_channel = journal.error('geogrid.get_point_epsg')
+
+    if lon >= 180.0:
+        lon = lon - 360.0
+
+    if lat >= 60.0:
+        return 3413
+    elif lat <= -60.0:
+        return 3031
+    elif lat > 0:
+        return 32601 + int(np.round((lon + 177) / 6.0))
+    elif lat < 0:
+        return 32701 + int(np.round((lon + 177) / 6.0))
+    else:
+        err_str = "'Could not determine EPSG for {0}, {1}'.format(lon, lat))"
+        error_channel.log(err_str)
+        raise ValueError(err_str)
+
+
+def generate_geogrids(all_bursts, geo_dict, dem):
+    dem_raster = isce3.io.Raster(dem)
+
+    # Unpack values from geocoding disctionary
+    epsg_dict = geo_dict['output_epsg']
+    x_start_dict = geo_dict['top_left']['x']
+    y_start_dict = geo_dict['top_left']['y']
+    x_spacing_dict = geo_dict['x_posting']
+    y_spacing_dict = geo_dict['y_posting']
+    x_end_dict = geo_dict['bottom_right']['x']
+    y_end_dict = geo_dict['bottom_right']['y']
+    x_snap_dict = geo_dict['x_snap']
+    y_snap_dict = geo_dict['y_snap']
+
+    geo_grids = {}
+    for common_id_bursts in all_bursts:
+        for burst in common_id_bursts:
+            burst_id = burst.burst_id
+
+            if burst_id in geo_grids:
+                continue
+
+            # Compute Burst epsg if not assigned in runconfig
+            if epsg_dict is None:
+                epsg_default = get_point_epsg(burst.center.y,
+                                              burst.center.x)
```

Comment:
You probably need an else for the case when epsg_dict i not None. Otherwise epsg_default is never defined and next line will fail.

Reply:
Yes, you are right. I should just compute the default epsg and then the next function `assign_epsg_check` checks if `epsg_dict` is None and if it is assigns the default value computed from the burst center coordinates